src/Sema.zig

```diff
@@ -9244,9 +9244,9 @@ fn cmpSelf(
     const resolved_type = sema.typeOf(casted_lhs);
     const runtime_src: LazySrcLoc = src: {
         if (try sema.resolveMaybeUndefVal(block, lhs_src, casted_lhs)) |lhs_val| {
-            if (lhs_val.isUndef()) return sema.addConstUndef(resolved_type);
+            if (lhs_val.isUndef()) return sema.addConstUndef(Type.initTag(.bool));
```

Comment:
```suggestion
            if (lhs_val.isUndef()) return sema.addConstUndef(Type.bool);
```

Reply:
Got it, thanks! You use `initTag(.bool)` one other place in Sema if you want to fix that up too at the same time, that's where I cargo culted this from.

Reply:
Will do :+1: